src/components/bottom-bar/bottom-bar-registry-manager.ts

```diff
@@ -0,0 +1,42 @@
+export interface IconComponent {
+  registerComponent(): any;
+  getSvgIcon(): any;
+  getComponentText(): any;
+}
+
+export class BottomBarRegistry {
+  private _registry: Map<string, any>;
+
+  constructor() {
+    this._registry = new Map();
+  }
+
+  public register(component: string, componentIcon: any): void {
+    if (!this._registry.get(component)) {
+      this._registry.set(component, componentIcon);
+    }
+  }
+
+  public unregister(component: string): void {
+    if (this._registry.get(component)) {
+      this._registry.delete(component);
+    }
+  }
+
+  public get registry(): Map<string, any> {
+    return this._registry;
```

Comment:
if register and unregister are public, why is registry also public ?

Reply:
register just adds the component to the map. in ui-managers when we need to add a component to the upper-bar, we need to get the component from the registry itself (the map).

Reply:
isn't that what get() does ? return a component from the map ?

Reply:
but you need to apply the `get()` on one of the items... to get the item you need the `registry`